GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexEntry.cs

```diff
@@ -122,7 +167,16 @@ public unsafe void ParsePath()
                     }
 
                     // We unrolled the final part calculation to after the loop, to avoid having to do a 0-byte check inside the for loop

```

Comment:
This comment is an indication of just how hot the loop above is. A single if statement had a measurable perf impact, so writing to memory will undoubtedly slow it down more than we'd like.